tapdb/assets_store.go

```diff
@@ -402,6 +415,15 @@ func (c cacheableBlockHeight) Size() (uint64, error) {
 	return 1, nil
 }
 
+// AssetBurn holds data related to a burn of an asset.
+type AssetBurn struct {
```

Comment:
Seems like we could use `tapfreighter.AssetBurn` instead, since we need to re-declare it there. Can move that declaration to this commit.